src/DIRAC/ConfigurationSystem/Client/Helpers/Resources.py

```diff
@@ -470,3 +488,50 @@ def getInfoAboutProviders(of=None, providerName=None, option='', section=''):
     return S_OK(gConfig.getValue(
         '%s/%sProviders/%s/%s/%s' % (gBaseResourcesSection, of, providerName,
                                      section, option)))
+
+
+def getProvidersForInstance(instance, providerType=None):
```

Comment:
I think that all these methods would be much easier and much faster if you were creating a subsection of `Resources` dedicated to privder and IDP. It would be much cleaner also.

Reply:
all new methods moved to DIRAC.Resources.IdProvider.Utilities